utils/disk_space_monitor.cc

```diff
@@ -76,7 +76,11 @@ future<> disk_space_monitor::poll() {
 }
 
 future<std::filesystem::space_info> disk_space_monitor::get_filesystem_space() {
-    return engine().file_system_space(_data_dir.native());
+    auto space = co_await engine().file_system_space(_data_dir.native());
+    if (_cfg.capacity_override()) {
+        space.capacity = _cfg.capacity_override();
```

Comment:
But now `free` and `available` are out of sync with capacity.
Maybe they need to be adjusted by (space.capacity - capacity_override) since this space is neither free nor available.

Also, should we allow capacity_override to be greater than the filesystem capacity?

Reply:
> But now `free` and `available` are out of sync with capacity. Maybe they need to be adjusted by (space.capacity - capacity_override) since this space is neither free nor available.

Right, will fix.

> 
> Also, should we allow capacity_override to be greater than the filesystem capacity?

Since the setting is for manual override and emergency situations, I think it's better to leave it unconstrained in case we need to compensate for bugs which would otherwise under-allocate the node.

Reply:
Agree